keyboards/handwired/lagrange/rules.mk

```diff
@@ -0,0 +1,63 @@
+# MCU name
+MCU = atmega32u4
+
+# Processor frequency.
+#     This will define a symbol, F_CPU, in all source code files equal to the
+#     processor frequency in Hz. You can then use this symbol in your source code to
+#     calculate timings. Do NOT tack on a 'UL' at the end, this will be done
+#     automatically to create a 32-bit value in your source code.
+#
+#     This will be an integer division of F_USB below, as it is sourced by
+#     F_USB after it has run through any CPU prescalers. Note that this value
+#     does not *change* the processor frequency - it should merely be updated to
+#     reflect the processor speed set externally so that the code can use accurate
+#     software delays.
+F_CPU = 16000000
+
+#
+# LUFA specific
+#
+# Target architecture (see library "Board Types" documentation).
+ARCH = AVR8
+
+# Input clock frequency.
+#     This will define a symbol, F_USB, in all source code files equal to the
+#     input clock frequency (before any prescaling is performed) in Hz. This value may
+#     differ from F_CPU if prescaling is used on the latter, and is required as the
+#     raw input clock is fed directly to the PLL sections of the AVR for high speed
+#     clock generation for the USB and other AVR subsections. Do NOT tack on a 'UL'
+#     at the end, this will be done automatically to create a 32-bit value in your
+#     source code.
+#
+#     If no clock division is performed on the input clock inside the AVR (via the
+#     CPU clock adjust registers or the clock division fuses), this will be equal to F_CPU.
+F_USB = $(F_CPU)
+
+# Bootloader
+#     This definition is optional, and if your keyboard supports multiple bootloaders of
+#     different sizes, comment this out, and the correct address will be loaded
+#     automatically (+60). See bootloader.mk for all options.
+BOOTLOADER = atmel-dfu
+
+# Interrupt driven control endpoint task(+60)
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+CFLAGS += -Wextra -Wno-unused-parameter -Wno-empty-body
```

Comment:
Is there a particular reason for this?

Reply:
I use `-Wextra` in all my code (and since you can't do much about unused parameters when dealing with APIs, I disable that warning).  It includes useful warnings.  For instance, after moving the debounce algorithm selection to its proper place, thus actually enabling it, I now get the following, warning:

```
quantum/debounce/sym_defer_pk.c: In function ‘update_debounce_counters_and_transfer_if_expired’:
tmk_core/common/timer.h:27:174: error: signed and unsigned type in conditional expression [-Werror=sign-compare]
 - (b))) : ((max == UINT16_MAX) ? ((uint16_t)((a) - (b))) : ((max == UINT32_MAX) ? ((uint32_t)((a) - (b))) : ((a) >= (
                                                                                                           ^
quantum/debounce/sym_defer_pk.c:81:21: note: in expansion of macro ‘TIMER_DIFF’
                 if (TIMER_DIFF(current_time, *debounce_pointer, MAX_DEBOUNCE) >= DEBOUNCE) {
 ```

Although harmless in this particular instance, such comparisons can harbor hard to spot bugs, so having such warnings enabled can help (especially since the code has already adopted a sensible, zero-tolerance `-Wall -Werror` policy).

I would therefore recommend adding `-Wextra`, disabling specific warnings as required, but I'll just remove it from my `rules.mk` for now, as it prevents compilation.